mw-webapp/src/service/auth/handleUserAuthState.ts

```diff
@@ -18,8 +20,10 @@ export const handleUserAuthState = (setUser: React.Dispatch<React.SetStateAction
     const userDocSnapshot = await getDoc(userDocRef);
 
     if (!userDocSnapshot.exists()) {
-      writeNewUserCredentials(currentUser);
+      await writeNewUserCredentials(currentUser);
     }
-    setUser(currentUser);
+
+    const currentUserPreview = await UserPreviewDAL.getUserPreview(currentUser.uid);
+    setUser(currentUserPreview);
```

Comment:
I think it will be better if auth function will not get additional requests, lets make it simple. Maybe it is better to return uuid and load user separately. In this case you should not even throw setUser callback and this functions become simple